Sources/Target/POSIX/ELFProcess.cpp

```diff
@@ -53,43 +53,56 @@ inline ErrorCode GetELFSharedLibraryInfoAddress(ELFProcess *process,
   PHDR ph;
   DYNSYM dynsym;
 
-  CHK(process->readMemory(process->loadBase(), &eh, sizeof(eh)));
-
-  if (eh.e_phnum == 0) {
+  // 1. Find the PHDR, number of entries, size of
+  //    each entry, and the loadBase of the binary.
+  //    Assumption: PHDR is attached to the ELF
+  //    header. This isn't enforced but it almost
+  //    always occurs in practice.
+  uint64_t phdrAddress = process->getAuxiliaryVectorValue(AT_PHDR);
+  size_t phdrEntrySize = process->getAuxiliaryVectorValue(AT_PHENT);
+  size_t phdrNumEntry = process->getAuxiliaryVectorValue(AT_PHNUM);
+  uint64_t loadBase = phdrAddress - sizeof(eh);
+  if (phdrAddress == 0 || phdrEntrySize == 0 || phdrNumEntry == 0) {
```

Comment:
I wouldn't mind having a `Debug` log here if one of these checks fails, to indicate that dynamic library info won't be available.